src/CommonDBRelation.php

```diff
@@ -2196,4 +2196,24 @@ public static function getMemberPosition(string $class): int
             return 0;
         }
     }
+
+    public static function getTypeName($nb = 0)
+    {
+        if (isset(static::$itemtype_2) && static::$itemtype_2 != 'itemtype') {
+            return static::$itemtype_2::getTypeName($nb);
+        } elseif (isset(static::$itemtype_1) && static::$itemtype_1 != 'itemtype') {
+            return static::$itemtype_1::getIcon();
```

Comment:
This seems wrong to return the icon.
Also not sure relation classes should automatically use the name of one of the linked types by default.